File: DeltaUtils/src/main/java/com/deltarobotics9351/pid/PIDCoefficients.java
/*
 * Created by FTC team Delta Robotics #9351
 *  Source code licensed under the MIT License
 *  More info at https://choosealicense.com/licenses/mit/
 */

package com.deltarobotics9351.pid;

/**
 * Class containing the PID Constants needed by some "drives"
 */
public class PIDCoefficients {

    /**
     * PID Coefficients
     */
    public double kP, kI, kD;

    /**
     * Constructor for PIDCoefficients class
     * @param kP the Proportional coefficient
     * @param kI the Integral coefficient
     * @param kD the Derivative coefficient
     */
    public PIDCoefficients(double kP, double kI, double kD){
        this.kP = kP;
        this.kI = kI;
        this.kD = kD;
    }

}

File: DeltaUtils/src/main/java/com/deltarobotics9351/pid/PIDCoefficients.kt
/*
 * Copyright (c) 2020 FTC Delta Robotics #9351 - Sebastian Erives
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.deltarobotics9351.pid

class PIDCoefficients(P: Double, I: Double, D: Double) {

    /**
     * PID Coefficients
     */
    var kP = 0.0
    var kI = 0.0
    var kD = 0.0

    /**
     * Constructor for PIDCoefficients class
     * @param kP the Proportional coefficient
     * @param kI the Integral coefficient
     * @param kD the Derivative coefficient
     */
    init {
        this.kP = P
        this.kI = I
        this.kD = D
    }

}